File: tests/fixtures/sample-python-project-negative/shared/utils/logger.py
# VIOLATION: style/deterministic/docstring-completeness
class Logger:
    # VIOLATION: style/deterministic/docstring-completeness
    def info(self, message: str, *args) -> None:
        # VIOLATION: code-quality/deterministic/console-log
        print(f"[INFO] {message}", *args)

    # VIOLATION: style/deterministic/docstring-completeness
    def error(self, message: str, *args) -> None:
        # VIOLATION: code-quality/deterministic/console-log
        print(f"[ERROR] {message}", *args)

    # VIOLATION: style/deterministic/docstring-completeness
    def warn(self, message: str, *args) -> None:
        # VIOLATION: code-quality/deterministic/console-log
        print(f"[WARN] {message}", *args)


logger = Logger()


# --- Logging pattern TPs (moved from synthetic batch files) ---

# VIOLATION: bugs/deterministic/logging-exception-outside-handler
def log_outside_handler(data):
    """logger.exception() called outside an except block."""
    logger.exception("Something went wrong")
    return data

File: reference/fixtures/sample-js-project-negative/shared/utils/src/logger.ts
// VIOLATION: architecture/deterministic/dead-module
// VIOLATION: code-quality/deterministic/unsafe-any-usage
const config: any = {};
config.level;

export const logger = {
  // VIOLATION: architecture/deterministic/dead-method
  // VIOLATION: code-quality/deterministic/unsafe-any-usage
  info: (message: string) => {
    const meta: any = {};
    meta.timestamp;
    // VIOLATION: code-quality/deterministic/console-log
    console.log(`[INFO] ${message}`);
  },
  // VIOLATION: code-quality/deterministic/unsafe-any-usage
  error: (message: string) => {
    const ctx: any = {};
    ctx.stack;
    console.error(`[ERROR] ${message}`);
  },
  warn: (message: string) => {
    console.warn(`[WARN] ${message}`);
  },
};
